### PR TITLE
Fix up pathing to be more straight forward

### PR DESCRIPTION
**Pull request by macdiesel**
@clintonb these changes will affect your open branch. We will start running all plays from the configuration directory root. This will remove much of the back-pathing that is currently in the code and will allow users to override the ansible configuration should it be necessary.

### CODE REVIEW

File: edxpipelines/patterns/stages.py [hunk 1]
```diff
@@ -542,7 +549,7 @@ def generate_run_migrations(pipeline,
         "stage": inventory_location.stage,
         "job": inventory_location.job,
         "src": FetchArtifactFile(inventory_location.file_name),
-        "dest": 'configuration'
+        "dest": constants.ARTIFACT_PATH
```
Comment (doctoryes) [inline edxpipelines/patterns/stages.py]: Incorrect dir before this PR?

File: edxpipelines/patterns/tasks.py [hunk 2]
```diff
@@ -178,31 +183,36 @@ def generate_run_migrations(job, sub_application_name=None, runif="passed"):
         The newly created task (gomatic.gocd.tasks.ExecTask)
 
     """
-    job.ensure_artifacts(set([BuildArtifact('target/unapplied_migrations.yml'),
-                              BuildArtifact('target/migration_output.yml')]))
+    job.ensure_artifacts(
+        set(
+            [BuildArtifact('{ARTIFACT_PATH}/migrations/'.format(ARTIFACT_PATH=constants.ARTIFACT_PATH))]
```
Comment (doctoryes) [inline edxpipelines/patterns/tasks.py]: Nit: Maybe:
`[BuildArtifact('{}/migrations/'.format(constants.ARTIFACT_PATH))]` ?

File: edxpipelines/patterns/tasks.py [hunk 3]
```diff
@@ -401,22 +415,22 @@ def generate_run_app_playbook(job, secure_dir, playbook_path, runif="passed", **
...
-            '-e @../../../{secure_dir}/ansible/vars/${{EDX_ENVIRONMENT}}-${{DEPLOYMENT}}.yml',
+            '--module-path=playbooks/library ',
+            '-i ../{artifact_path}/ansible_inventory '
+            '-e @../{artifact_path}/launch_info.yml',
+            '-e @../{secure_dir}/ansible/vars/${{DEPLOYMENT}}.yml',
+            '-e @../{secure_dir}/ansible/vars/${{EDX_ENVIRONMENT}}-${{DEPLOYMENT}}.yml',
         ]
     )
-    command = command.format(secure_dir=secure_dir)
+    command = command.format(secure_dir=secure_dir, artifact_path=constants.ARTIFACT_PATH)
```
Comment (doctoryes) [inline edxpipelines/patterns/tasks.py]: Nit: Should probably standardize on `artifact_path` or `ARTIFACT_PATH` for the string format replacement.

File: edxpipelines/patterns/tasks.py [hunk 4]
```diff
@@ -428,7 +442,7 @@ def generate_run_app_playbook(job, secure_dir, playbook_path, runif="passed", **
                 '-c',
                 command
             ],
-            working_dir="configuration/playbooks/continuous_delivery/",
+            working_dir=constants.PUBLIC_CONFIGURATION_DIR,
```
Comment (doctoryes) [inline edxpipelines/patterns/tasks.py]: Definitely makes more sense to run these from the top-level cloned repo dir, BTW.

File: edxpipelines/patterns/tasks.py [hunk 5]
```diff
@@ -92,10 +94,10 @@ def generate_create_ami(job, runif="passed", **kwargs):
         [
             'ansible-playbook',
             '-vvvv',
-            '--module-path=../library',
+            '--module-path=playbooks/library',
             '-i "localhost,"',
             '-c local',
-            '-e @../../../target/launch_info.yml',
+            '-e @../target/launch_info.yml',
```
Comment (doctoryes) [inline edxpipelines/patterns/tasks.py]: Why use `ARTIFACT_PATH` above and `target` here?

**Comment by doctoryes** [comment]
First pass complete.

File: edxpipelines/patterns/stages.py (hunk 1 above)
Comment (macdiesel) [inline edxpipelines/patterns/stages.py]: not necessary, we could put this file anywhere as long as we pass the correct path in to the ansible job. What I did while going through this module to fix pathing was to ensure that we are always putting the artifacts in the same place. 

This was a change for the sake of consistency.

File: edxpipelines/patterns/stages.py [hunk 6]
```diff
@@ -565,15 +572,15 @@ def generate_run_migrations(pipeline,
...
     }
     job.add_task(FetchArtifactTask(**artifact_params))
 
     # The SSH key used to access the EC2 instance needs specific permissions.
     job.add_task(
         ExecTask(
             ['/bin/bash', '-c', 'chmod 600 {}'.format(instance_key_location.file_name)],
-            working_dir='configuration'
+            working_dir='target'
```
Comment (macdiesel) [inline edxpipelines/patterns/stages.py]: This should be constants.ARTIFACT_PATH.

File: edxpipelines/patterns/stages.py (hunk 1 above)
Comment (doctoryes) [inline edxpipelines/patterns/stages.py]: 👍

**Comment by macdiesel** [comment]
@doctoryes Things are even more consistently consistent constantly in this pr.

File: edxpipelines/patterns/tasks.py (hunk 3 above)
Comment (macdiesel) [inline edxpipelines/patterns/tasks.py]: done

**Comment by doctoryes** [comment]
👍 🚢 🇮🇹

**Pull request merged**